torch/csrc/api/include/torch/mps.h

```diff
@@ -15,7 +24,19 @@ bool TORCH_API is_available();
 void TORCH_API manual_seed(uint64_t seed);
 
 /// Waits for all streams on a MPS device to complete.
+/// See this link for more info:
+/// https://developer.apple.com/documentation/metal/mtlcommandbuffer/1443039-waituntilcompleted
```

Comment:
Thanks for adding this link but this still feels quite confusing to me.
For example the names "stream" and "command buffer"/"dispatch queue" are used without defining the difference between them.


At a high level I would expect that:
- You have links on what these data structures are in the mps docs
- Have explanation on how these structures are linked to the concepts that exist on at::device
- For the function API, you either need to say: (defined based off MPS function) "this function is equivalent to waitUntilCompleted which is described in the mps doc here" or (defined based off pytorch concepts) "this function is synchronizing the GPU by waiting for all ops in all streams to be done". I think the second one is best as it improves consistency within PT and makes it simpler to understand what this API is doing for other PT maintainers.
- If you need a whole Note on the website or in a specific file that maps MPS names to PT names, we should do that and link it from the code. Within code, we use `Note [Some Name]` to write longer notes that are linked from multiple places for example https://github.com/search?q=repo%3Apytorch%2Fpytorch+%2FNote+%5C%5BMove+construction+for+RAII+guards+is+tricky%5C%5D%2F&type=code

Reply:
Thanks @albanD for the suggestions. Since these comments for C++ APIs and the ones in `mps.rst` for python bindings need a good refining/scrubbing, if you don't mind I'll update both of them in a separate PR soon and will add you to the review (a PR just for API docs). Sounds good to you?

Reply:
>  the ones in mps.rst for python bindings

This is for the python side yes. I think a Note in the C++ code would be very valuable though for these lower level APIs!
Or if they all make it to public APIs, then having them nicely documented at the python level (with c++ pointing to the python doc) sounds good.

I'm fine with doing this in a follow up. But I don't feel like I can properly review PRs like this one without such a doc. So I would expect this before I can help with more reviews.

Reply:
Sure, I'll talk to Kulin and will follow up on that with you.

Reply:
I updated the comments for `torch::mps::synchronize()`.